xapian-letor/api/featuremanager.cc

```diff
@@ -1,174 +1,109 @@
+/** @file featuremanager.cc
+ * @brief FeatureManager class
+ */
+/* Copyright (C) 2011 Parth Gupta
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ */
+#include <xapian.h>
+#include <config.h>
+
 #include "xapian-letor/featuremanager.h"
 #include "xapian-letor/featurevector.h"
 #include "xapian-letor/ranklist.h"
+#include "featuremanager_internal.h"
+
 #include <cstring>
 #include <cstdlib>
 #include <fstream>
 
 #include <map>
+#include <string>
 
+using namespace std;
 
+namespace Xapian {
 
-using namespace Xapian;
-
-using namespace std;
+FeatureManager::FeatureManager() : internal(new FeatureManager::Internal) 
+{ 
+}
 
+FeatureManager &
+FeatureManager::operator=(const FeatureManager & o)
+{
+    internal = o.internal;
+    return *this;
+}
 
+FeatureManager::FeatureManager(const FeatureManager & o) : internal(o.internal) 
+{ 
+}
 
-FeatureManager::FeatureManager() {
+FeatureManager::~FeatureManager() 
+{ 
 }
 
 std::string
-FeatureManager::getdid(const Document &doc) {
-    string id="";
-    string data = doc.get_data();
-    string temp_id = data.substr(data.find("url=", 0), (data.find("sample=", 0) - data.find("url=", 0)));
```

Comment:
It's worth wrapping these lines to try to get the bulk of them within 80 characters. (It's not always possible easily, but there's some straightforward approaches here that will work.)